### ISSUE TITLE
Error while running get_python script results in incomplete configuration

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#13992]
**Describe the bug**
While running the get_python script the following error occurs:
```
Collecting gitdb<5,>=4.0.1
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    gitdb<5,>=4.0.1 from https://files.pythonhosted.org/packages/21/a6/35f83efec687615c711fe0a09b67e58f6d1254db27b1013119de46f450bd/gitdb-4.0.10-py3-none-any.whl (from gitpython==3.1.30->-r C:\o3de\python/requirements.txt (line 48)
```
Running the configure command after that prints numerous errors and fails.
Please note that the issue occurs for Windows, Linux and Mac.

Refer to the attached log for more information.

The issue is related to the following commit - [6daf594](https://github.com/o3de/o3de/commit/6daf594efde8540bc61ec407308cad29bb660962).

**Steps to reproduce**
1. Obtain the latest Development from Github.
2. Open a command prompt and change to the directory where O3DE is set up and run the get_python script:
`python\get_python.bat`

**Expected behavior**
The script runs without errors.

**Actual behavior**
An error occurs while running the script resulting in incomplete configuration.

**Found in Branch**
Development ([228fd98](https://github.com/o3de/o3de/commit/228fd98fd8eed3c3315594c4177830677569fa54))

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
([228fd98](https://github.com/o3de/o3de/commit/228fd98fd8eed3c3315594c4177830677569fa54))

**Desktop/Device**
- Device: PC
- OS: Windows
- Version: 10
- CPU: AMD Ryzen 5 3600
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[GetPython&Configure.txt](https://github.com/o3de/o3de/files/10390161/GetPython.Configure.txt)

**Comment by allisaurus** [comment]
Potentially resolved by: https://github.com/o3de/o3de/pull/14004

**Comment by LB-KacperKapusta** [comment]
Issue verified as fixed in Development ([7dee349](https://github.com/o3de/o3de/commit/7dee34909edc04fd872250cb383ab169926400b9)). Get_python script and configure complete without errors. 
Closing the issue.
